iocage_lib/ioc_fstab.py

```diff
@@ -61,41 +63,155 @@ def __init__(self, uuid, action, source, destination, fstype, fsoptions,
+                        f'Destination: {self.dest} must include '
+                        f'jail\'s mountpoint! ({jail_root})'
+                    )
+                    break
+            else:
+                if jail_root not in destination:
+                    verrors.append(
+                        f'Destination: {destination} does not include '
+                        f'jail\'s mountpoint! ({jail_root})'
+                    )
+                    missing_root = True
+
+            if not source.is_dir():
+                verrors.append(f'Source: {source} does not exist!')
+            if not source.is_absolute():
+                verrors.append(f'Source: {source} must use an absolute path!')
+
+            if not missing_root:
+                if not dest.is_dir():
+                    verrors.append(f'Destination: {dest} does not exist!')
+                if not dest.is_absolute():
+                    verrors.append(
+                        f'Destination: {dest} must use an absolute path!'
+                    )
+
+            if not dump.isdecimal():
+                verrors.append(
+                    f'Dump: {dump} must be a digit!'
+                )
+            if len(dump) > 1:
+                verrors.append(
+                    f'Dump: {dump} must be one digit long!'
+                )
+            if not _pass.isdecimal():
+                verrors.append(
+                    f'Pass: {_pass} must be a digit!'
+                )
+            if len(_pass) > 1:
+                verrors.append(
+                    f'Pass: {_pass} must be one digit long!'
+                )
+            dests.append(destination)
+
+        if verrors:
+            iocage_lib.ioc_common.logit({
+                'level': 'EXCEPTION',
+                'message': verrors
+            },
+                _callback=self.callback,
+                exception=iocage_lib.ioc_exceptions.ValidationFailed
+            )
+
+        return dests
 
     def __fstab_add__(self):
         """Adds a users mount to the jails fstab"""
-        with open(f"{self.iocroot}/jails/{self.uuid}/fstab", "r") as fstab:
-            with iocage_lib.ioc_common.open_atomic(
-                    f"{self.iocroot}/jails/{self.uuid}/fstab",
-                    "w") as _fstab:
-                # open_atomic will empty the file, we need these still.
+        with iocage_lib.ioc_common.open_atomic(
+                f'{self.iocroot}/jails/{self.uuid}/fstab',
+                'w'
+        ) as fstab:
+            for line in self.fstab:
+                fstab.write(f'{line}\n')
 
-                for line in fstab.readlines():
-                    _fstab.write(line)
-
-                date = datetime.datetime.utcnow().strftime("%F %T")
-                _fstab.write(f"{self.mount} # Added by iocage on {date}\n")
+            date = datetime.datetime.utcnow().strftime("%F %T")
+            fstab.write(f"{self.mount} # Added by iocage on {date}\n")
```

Comment:
Should the contents of `self.fstab` be updated if the object is to be used somewhere else as well ?

Reply:
That would be the case on reading too as we went to the end for each of the reads on it. But in this case, only one action can happen at a time, it's shared to reduce code, not because it can be used multiple times.